tests/application/static_site/test_views.py

```diff
@@ -865,6 +865,69 @@ def test_later_version_shows_links_to_earlier_versions(self, test_app_client, lo
         measure_1_0_links = page.find_all("a", attrs={"href": measure_1_0_url})
         assert len(measure_1_0_links) == 0
 
+    def test_version_with_factual_error(self, test_app_client, logged_in_rdu_user):
+
+        # GIVEN a 1.0 measure which contains factual errors
+        measure = MeasureFactory()
+
+        measure_1_0 = MeasureVersionFactory.create(measure=measure, status="APPROVED", version="1.0")
+
+        MeasureVersionFactory.create(
+            measure=measure,
+            status="APPROVED",
+            version="1.1",
+            update_corrects_data_mistake=True,
+            update_corrects_measure_version=measure_1_0.id,
+        )
+
+        measure_1_0_url = url_for(
+            "static_site.measure_version",
+            topic_slug=measure_1_0.measure.subtopic.topic.slug,
+            subtopic_slug=measure_1_0.measure.subtopic.slug,
+            measure_slug=measure_1_0.measure.slug,
+            version=measure_1_0.version,
+        )
+
+        # WHEN we get the 1.0 measure page
+        resp = test_app_client.get(measure_1_0_url)
+
+        assert resp.status_code == 200
+
+        page = BeautifulSoup(resp.data.decode("utf-8"), "html.parser")
+
+        # THEN it should contain a banner explaining that the page contains an error
+        assert page.find("h2", string="This page contains a factual mistake")
+        assert "Details can be found on the corrected page." in page.get_text()
```

Comment:
Can we be more specific with `assert "Details can be found on the corrected page." == page.get_text(strip=True)`

Reply:
@samuelhwilliams that wouldn't work? The intention here was just to assert that the text appears somewhere on the page, rather than baking in assumptions about classes or IDs. Possibly ideally we'd assert that it appears immediately _after_ the "This page contains a factual mistake" heading?

Reply:
Hmmmm. Maybe it's fine. I would love to have some sort of solid guide on how to check for things in pages in a reliable but not over-burdensome way.

Reply:
Me too! 😄